pkg/engine/task/podexec/pod_exec.go

```diff
@@ -185,17 +186,14 @@ func untarFile(fs afero.Fs, r io.Reader, fileName string) error {
 		switch header.Typeflag {
 		// if it's a file create it
 		case tar.TypeReg:
-			f, err := fs.OpenFile(target, os.O_CREATE|os.O_RDWR, os.FileMode(header.Mode))
+			buf, err := ioutil.ReadAll(tr)
 			if err != nil {
 				return err
 			}
-			defer f.Close() //nolint:errcheck
 
-			// copy over contents
-			if _, err := io.Copy(f, tr); err != nil {
+			if err := afero.WriteFile(fs, target, buf, os.FileMode(header.Mode)); err != nil {
```

Comment:
This is quite a bit of change, right?

In the old code we open the file descriptor, and then have a copy call. In the second, we read the whole file into memory and then write it back into a new file.

I assume that the `io.Copy` that was used before doesn't read the full file but copies the data in chunks?

Maybe there's a way to replicate this behavior in the new code?

Reply:
I forget to add comments why I think that this is necessary. By using `io.Copy` on files from an archive, we could have a DoS in case of a zip bomb. Because in those cases, data will be written until the file system complains, in worst case because it is full. I'm trying to work around this by using a buffer. If a large file would be read, the buffer will become too large and crash the manager. Though, while thinking about it again, I don't like that approach either, because the zip bomb would still work, just at a different level. An alternative approach would be to set a size limit that is accumulated over all written files, but what would be a good limit? 4GB maybe?